examples/hyper.rs

```diff
@@ -0,0 +1,89 @@
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Request, Response, Server,
+};
+use prometheus_client::{encoding::text::encode, metrics::counter::Counter, registry::Registry};
+use std::{
+    future::Future,
+    io,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    pin::Pin,
+    sync::Arc,
+    time::Duration,
+};
+use tokio::signal::unix::{signal, SignalKind};
+
+lazy_static::lazy_static! {
+    pub static ref EXAMPLE_COUNTER: Counter = Counter::default();
+}
+
+#[tokio::main]
+async fn main() {
+    let registry = register_metrics();
+
+    // Spawn a server to serve the OpenMetrics endpoint.
+    let metrics_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 8001);
+    tokio::task::spawn(start_server(metrics_addr, registry));
+
+    loop {
+        EXAMPLE_COUNTER.inc();
+        tokio::time::sleep(Duration::from_secs(1)).await;
+    }
```

Comment:
This is a neat idea, though for the sake of consistency with the other examples, I would prefer showcasing a standard HTTP request counter in this example.

See tide example:

https://github.com/prometheus/client_rust/blob/12c1de56116d803c709ed6584c263353727221fa/examples/tide.rs#L47-L57

I think an example should showcase a single thing only. (Though it is hard where the boundaries of a thing are.) I am fine for something like the above to be shown in a separate example.